### PR TITLE
Feature/ma doxygen doc

### LINKED ISSUE #52: Doxygen Doc - Pipipline
A lot of the code is Doxygen compatible commented. 
Setup a job in the pipeline that generates and artifacts Doxygen HTML output.

### PR DESCRIPTION
**Pull request by MAhmedDev**
Generate Doxygen documentation using docker 
Generate locally
from the root repo directory run 
```
make doxy-doc
```
doxygen outputs to Doxygen/html
Open the index.html

Generates documents in the pipeline unfortunately GitHub cannot serve HTML pages like GitLab
Triggers: master and develop branches

### CODE REVIEW

**Comment by MAhmedDev** [comment]
Closes #52

File: Makefile [hunk 1]
```diff
@@ -12,3 +12,6 @@ push: ## Push image
 
 pull: ## Pull image 
 	DOCKER_BUILDKIT=1 docker-compose -f docker-compose.yml pull
+
+doxy-doc: ## Generate Doxygen documentation
+	docker run --rm -it -v $(PWD):/stm32-app $(shell docker build -q -f Doxygen/Dockerfile .) /bin/sh -c "doxygen Doxygen/config/Doxyfile"
```
Comment (Niuslar) [inline Makefile]: I think adding quotes to the path fixes the error when the folder name has spaces. 
```"$(PWD):/stm32-app"```. 

I tried using that and the command runs without errors.

Comment (Niuslar) [inline Makefile:17]: I think adding quotes to the path fixes the error I got when the folder has spaces. 
```"$(PWD):/stm32-app"```. 

I tried using that and the command runs without errors.

Comment (MAhmedDev) [reply Makefile]: Thanks fixed.

**Review by Niuslar — APPROVED** [review]

**Pull request merged**